scripts/check_logs.sh

```diff
@@ -1,5 +1,10 @@
 #!/bin/bash
 
+GREP_BIN=grep
```

Comment:
Why was this needed?

We also have a `find`, do we need to do the same with that then?

Reply:
This is required so that users can provide their own grep binary. BSD grep and GNU grep differ quite a bit; so rather than trying to rewrite the tests/checks to achieve compatibility, this enables the caller to pass a compatible grep binary insteadf. On macos I can install GNU grep (which doesn't replace the system grep) and set the path when executing any of these grep based tests/checks.